ldap/servers/slapd/back-ldbm/idl_new.c

```diff
@@ -566,46 +567,22 @@ idl_new_range_fetch(
             idl_new_exceeds_allidslimit(count, ai, allidslimit)) {
             idl->b_nids = 1;
             idl->b_ids[0] = ALLID;
-            ret = DB_NOTFOUND; /* fool the code below into thinking that we finished the dups */
+            ret = DBI_RC_NOTFOUND; /* fool the code below into thinking that we finished the dups */
             break;
         }
 #endif
-        ret = cursor->c_get(cursor, &cur_key, &data, DB_NEXT_DUP | DB_MULTIPLE);
-        if (saved_key != cur_key.data) {
-            /* key was allocated in c_get */
-            slapi_ch_free(&saved_key);
-            saved_key = cur_key.data;
-        }
+        ret = dblayer_cursor_bulkop(&cursor, DBI_OP_NEXT_KEY, &cur_key, &bulkdata);
```

Comment:
idl_new handle duplicate keys, shouldn't it be DBI_OP_NEXT_DATA (i.e DB_NEXT_DUP | DB_MULTIPLE) ?

Reply:
Good catch !!!